docs/strings.rst

```diff
@@ -28,6 +28,24 @@ will be treated as variable-length strings, marked as UTF-8 and ASCII respective
 Numpy bytes arrays (``'S'`` dtypes) make fixed-length strings.
 You can use :func:`.string_dtype` to explicitly specify any HDF5 string datatype.
 
+::
+
+    string_data = ["varying", "sizes", "of", "strings"]
+
+    # Variable length strings (implicit)
+    f['vlen_strings1'] = string_data
+
+    # Variable length strings (explicit)
+    ds = f.create_dataset('vlen_strings2', shape=4, dtype=h5py.string_dtype())
```

Comment:
It'd be nice if there were some examples with `data=` keyword argument - those are the ones that got me confused.

Reply:
Passing `data=` should be equivalent to the implicit case (`f[name] =`) if you don't pass `dtype=`, or to the explicit case if you do. I'm showing the explicit dtype case as two lines to illustrate that it lets you create the dataset before you have all the data.

I'd rather not make this example longer by showing more possible ways to do the same thing. The explanation at [creating datasets](https://docs.h5py.org/en/stable/high/dataset.html#creating-datasets) could probably be improved as well.